internal/config/config.go

```diff
@@ -548,11 +628,62 @@ func validInstanceTypeForProvider(insType string, acceptNonCVM bool, provider cl
 			}
 		}
 		return false
+	case cloudprovider.GCP:
+		for _, instanceType := range instancetypes.GCPInstanceTypes {
+			if insType == instanceType {
+				return true
+			}
+		}
+		return false
 	default:
 		return false
 	}
 }
 
+// checkIfAWSInstanceTypeIsValid checks if an AWS instance type passed as user input is in one of the many supported instance families or single instances.
+func checkIfAWSInstanceTypeIsValid(userInput string) bool {
+	/*
+		Check for supported instances with the following characterics:
+		- must be an instance built on "Nitro System"
+		- at least 4 vCPUs (contains xlarge in string, everything > 4 vCPUs is a multiplicator of xlarge)
+		- not Graviton (arm64) based, as we do not have ARM images yet
+		Graviton instances have 'g' in their name, but not at the very beginning (these are 'G' family instances)
+
+		If we do support Graviton instances in the future, this check can be easily removed but we would need to make sure that the images are built for arm64.
+	*/
+
+	// Check if user or code does anything weird and tries to pass multiple strings as one
+	if strings.Contains(userInput, " ") {
+		return false
+	}
+	if strings.Contains(userInput, ",") {
+		return false
+	}
+	if strings.Contains(userInput, ";") {
+		return false
+	}
+
+	userInputLowercase := strings.ToLower(userInput)
+	family := strings.Split(userInputLowercase, ".")[0]
+	isGraviton := strings.LastIndex(family, "g") > 0
+	hasEnoughVCPUs := strings.Contains(userInputLowercase, "xlarge")
```

Comment:
`m6id.metal` is part of the `AWSSupportedInstanceTypesOrFamilies` list. Is this type allowed?
What could work here is to explicitly disallow `medium`, `micro` `large` and so on.

Reply:
Guess I forgot checking for metal types here. But just adding `metal` in addition to `xlarge` should work better than disallowing the others I guess.